Client pacnis if p256-pubkey.pem is the wrong key

As of commit 9000ad94d8fa058ad222350472008c3ad8883f3e, If the client is verifying signatures using a public key `p256-pubkey.pem` that does not match the server signing key `p256-key.pem`, the client panics instead of displaying invalid signature error. Below is the output.

**Repro:**

1. Run `./scripts/prepare_server.sh` and configure both server and client to run on the same machine (this is the default config).
2. Configure the client to read keys from `genfiles/`.
3. Run
    ```
    ./key-transparency-client post <email> -d '{"app1": "dGVzdA=="}' --config=./.key-transparency.yaml --verbose
    ```
4. Configure the client to read `p256-pubkey.pem` from `testdata/`.
5. Run the same command in step 3.

**Output:**

```
2016/11/24 16:13:41 Got current entry...
2016/11/24 16:13:41 ✓ Commitment verified.
2016/11/24 16:13:41 ✓ VRF verified.
2016/11/24 16:13:41 ✓ Sparse tree proof verified.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x51c117]

goroutine 1 [running]:
panic(0x93dbe0, 0xc4200120b0)
	/usr/lib/google-golang/src/runtime/panic.go:507 +0x1b3
github.com/google/key-transparency/core/signatures.Verifier.Verify(0x972280, 0xc42042da40, 0xc420159900, 0x8, 0x96aa20, 0xc4202e2580, 0x0, 0xc4203ce540, 0x22)
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/core/signatures/signatures.go:210 +0x37
github.com/google/key-transparency/core/client/kt.(*Verifier).VerifyGetEntryResponse(0xc420428570, 0x7fad4bb04800, 0xc42017c660, 0x7ffe7bd4150c, 0x15, 0xc42017c6c0, 0x0, 0xc42017c6c0)
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/core/client/kt/verify.go:108 +0x585
github.com/google/key-transparency/cmd/key-transparency-client/grpcc.(*Client).Update(0xc4203fc1e0, 0x7fad4bb04800, 0xc42017c660, 0x7ffe7bd4150c, 0x15, 0xc42003a3a8, 0x0, 0x0, 0x0, 0xeaf7a0, ...)
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/cmd/key-transparency-client/grpcc/grpc_client.go:188 +0x17d
github.com/google/key-transparency/cmd/key-transparency-client/cmd.glob..func4(0xe7b8a0, 0xc42012f220, 0x1, 0x5, 0x0, 0x0)
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/cmd/key-transparency-client/cmd/post.go:76 +0x3f4
github.com/google/key-transparency/vendor/github.com/spf13/cobra.(*Command).execute(0xe7b8a0, 0xc42012f1d0, 0x5, 0x5, 0xe7b8a0, 0xc42012f1d0)
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/vendor/github.com/spf13/cobra/command.go:632 +0x23e
github.com/google/key-transparency/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xe7bac0, 0x0, 0x0, 0x7)
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/vendor/github.com/spf13/cobra/command.go:722 +0x367
github.com/google/key-transparency/vendor/github.com/spf13/cobra.(*Command).Execute(0xe7bac0, 0x0, 0x0)
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/vendor/github.com/spf13/cobra/command.go:681 +0x2b
github.com/google/key-transparency/cmd/key-transparency-client/cmd.Execute()
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/cmd/key-transparency-client/cmd/root.go:75 +0x31
main.main()
	/usr/local/google/home/cesarghali/go/src/github.com/google/key-transparency/cmd/key-transparency-client/main.go:20 +0x14
```

*Note:* line numbers in the output trace are based on commit 9000ad94d8fa058ad222350472008c3ad8883f3e.

--- a/core/signatures/signatures.go
+++ b/core/signatures/signatures.go
@@ -40,6 +40,8 @@ var (
 	ErrWrongKeyType = errors.New("not an ECDSA key")
 	// ErrPointNotOnCurve occurs when a public key is not on the curve.
 	ErrPointNotOnCurve = errors.New("point is not on the P256 curve")
+	// ErrMissingSig occurs when the Verify function is called with a nil signature.
+	ErrMissingSig = errors.New("missing signature")
 	// ErrWrongHashAlgo occurs when a hash algorithm other than SHA256 was specified.
 	ErrWrongHashAlgo = errors.New("not the SHA256 hash algorithm")
 	// ErrWrongSignatureAlgo occurs when a signature algorithm other than ECDSA was specified.
@@ -207,6 +209,9 @@ func NewVerifier(pk crypto.PublicKey) (*Verifier, error) {
 
 // Verify checks the digital signature associated applied to data.
 func (s Verifier) Verify(data interface{}, sig *ctmap.DigitallySigned) error {
+	if sig == nil {
+		return ErrMissingSig
+	}
 	if sig.HashAlgorithm != ctmap.DigitallySigned_SHA256 {
 		return ErrWrongHashAlgo
 	}

